skins/laika/src/store/util.ts

```diff
@@ -11,12 +12,21 @@ export const action = buildActionOrMutationName;
 export const mutation = buildActionOrMutationName;
 
 export const Helper = {
-	inputIsValid: ( value: string, pattern: string ) => {
+	inputIsValid: ( value: string, pattern: string, isOptional?: boolean ) => {
+		if ( isOptional && Helper.isEmpty( value ) ) {
+			return Validity.VALID;
+		}
 		if ( pattern === null ) {
 			return value !== '' ? Validity.VALID : Validity.INVALID;
 		}
+		if ( Helper.isEmpty( value ) ) {
+			return Validity.INVALID;
+		}
 		return new RegExp( pattern ).test( value ) ? Validity.VALID : Validity.INVALID;
 	},
+	isEmpty( value: string ) {
+		return value === undefined || value === null || value === '';
```

Comment:
With TypeScript that's technically impossible. We're always passing a string, so all the check should do is compare the value to an empty string.

Reply:
I think it allows undefined and null to be passed anyway, that's why I had to add it. The subject drop down can be in a null state and I added undefined just to be sure.

Reply:
We have strict mode set to true in the TS compiler options, so strict null checks is also on -> you can't pass undefined and null to a string.

Reply:
If you actually want value here to be a string or null, we need a union type.